### DISCUSSION TITLE
Some suggestions for improving containerlab

### DISCUSSION THREAD

**Discussion by ltphung** [srl-labs/containerlab#2282] [Ideas]
### 1. Auto tag version number when manual build, or better guide for manual build ? ###
the var section in file  cmd/version.go :
var (
        version = "0.0.0"      << version is set 0.0.0 so when manual build will have issue with some tools, example netlab
        commit  = "none"
        date    = "unknown"
)

### Example output below, when "git clone" follow by "go build" ###
root@ubuntu22netlab:~/clab_build/containerlab# ./containerlab version
  ____ ___  _   _ _____  _    ___ _   _ _____ ____  _       _     
 / ___/ _ \| \ | |_   _|/ \  |_ _| \ | | ____|  _ \| | __ _| |__  
| |  | | | |  \| | | | / _ \  | ||  \| |  _| | |_) | |/ _` | '_ \ 
| |__| |_| | |\  | | |/ ___ \ | || |\  | |___|  _ <| | (_| | |_) |
 \____\___/|_| \_| |_/_/   \_\___|_| \_|_____|_| \_\_|\__,_|_.__/ 

    version: 0.0.0
     commit: none
       date: unknown
     source: https://github.com/srl-labs/containerlab
 rel. notes: https://containerlab.dev/rn/0.0/
 
 ##################################################################################
### 2. Copy /nodes dir and subdir/files... to /etc/containerlab/, similar to lab-examples/
Reason:
- Makes it easier for users to change default/startup configurations for nodes  --> this prevents the need to track down account credentials, or the need to rebuild the containerlab executable, or recreate Docker images to support new user default config for some nodes.

Sorry for my bad English :)

**Comment by hellt** [comment]
Hi!

for 1) you should make use of `make build` that will incorporate the latest commit id, but you should understand that this commit id is the one that is currently recorded, if you do `make build` over the changes that you haven't committed locally, then the prev commit will be used.

for 2) this is not how things work... We embed artifacts into the containerlab binary instead of loading them dynamically from the files on a filesystem

**Comment by ltphung** [comment]
Hi hellt !
for 1) you should make use of make build that will incorporate the latest commit id, but you should understand that this commit id is the one that is currently recorded, if you do make build over the changes that you haven't committed locally, then the prev commit will be used.
     - Honestly, I'm not quite sure what you mean :D! I followed the instructions in [this section](https://containerlab.dev/install/#from-source) and assumed it included **git clone** followed by **go build** (sorry if I'm mistaken; I’m really new to DevOps stuff ). 
     
for 2) this is not how things work... We embed artifacts into the containerlab binary instead of loading them dynamically from the files on a filesystem
   - Sure, I just think loading them dynamically might make it easier for lots of network people, like me, who are just starting to learn about automation and DevOps. :)
   
   Again, containerlab is fun to play with !
